django/cantusdb_project/cantusindex.py

```diff
@@ -0,0 +1,150 @@
+    sorted_suggestions: list = sorted(
+        all_suggestions, key=sort_by_occurrences, reverse=True
+    )
+    trimmed_suggestions: list = sorted_suggestions[:number_of_suggestions]
+
+    suggested_chants: list[dict] = []
+    for suggestion in trimmed_suggestions:
+        cantus_id: str = suggestion["cid"]
+        occurrences: int = int(suggestion["count"])
+        suggested_chants.append(get_suggested_chant(cantus_id, occurrences))
+
+    # filter out Cantus IDs where get_suggested_chant timed out
+    filtered_suggestions: list[dict] = [
+        sugg for sugg in suggested_chants if sugg is not None
+    ]
+
+    return filtered_suggestions
+
+
+def get_suggested_chant(
+    cantus_id: str, occurrences: int, timeout: float = DEFAULT_TIMEOUT
+) -> Optional[dict]:
+    """Given a Cantus ID and a number of occurrences, query one of Cantus Index's
+    APIs for information on that Cantus ID and return a dictionary
+    containing a full text, an incipit, the ID of that Cantus ID's genre, and
+    the number of occurrences for that Cantus ID
+
+    (Number of occurrences: this function is used on the Chant Create page,
+    to suggest Cantus IDs of chants that might follow a chant with the Cantus ID
+    of the most recently created chant within the current source. Number of occurrences
+    is provided by Cantus Index's /nextchants API, based on which chants follow which
+    other chants in existing manuscripts)
+
+    Args:
+        cantus_id (str): a Cantus ID
+        occurrences (int): the number of times chants with this Cantus ID follow chants
+            with the Cantus ID of the most recently created chant.
+
+    Returns:
+        Optional[dict]: A dictionary with the following keys:
+            - "cantus_id"
+            - "occurrences"
+            - "fulltext"
+            - "incipit"
+            - "genre_id"
+            ...but if get_json_from_ci_api timed out, returns None instead
+    """
+    endpoint_path: str = f"/json-cid/{cantus_id}"
+    json: dict = get_json_from_ci_api(endpoint_path, timeout=timeout)
+
+    if not json:  # mostly, in case of a timeout within get_json_from_ci_api
+        return None
+
+    fulltext: str = json["info"]["field_full_text"]
+    incipit: str = " ".join(fulltext.split(" ")[:5])
+    genre_name: str = json["info"]["field_genre"]
+    genre_id: Optional[int] = None
+    try:
+        genre_id = Genre.objects.get(name=genre_name).id
+    except Genre.DoesNotExist:
+        pass
+
+    clean_cantus_id = cantus_id.replace(".", "d").replace(":", "c")
+    #                                        "d"ot             "c"olon
+    return {
+        "cantus_id": cantus_id,
+        "occurrences": occurrences,
+        "fulltext": fulltext,
+        "incipit": incipit,
+        "genre_name": genre_name,
+        "genre_id": genre_id,
+        "clean_cantus_id": clean_cantus_id,
+    }
+
+
+def get_json_from_ci_api(
+    path: str, timeout: int = DEFAULT_TIMEOUT
```

Comment:
```suggestion
    path: str, timeout: float = DEFAULT_TIMEOUT
```

`DEFAULT_TIMEOUT` is a float.